pkg/operator/controller/externaldns/pod_test.go

```diff
@@ -0,0 +1,264 @@
+		},
+		{
+			name: "invalid regex exclude filter",
+
+			domainInput: []v1alpha1.ExternalDNSDomain{
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeRegex,
+						Pattern:   pointer.StringPtr(`(.*]\.abc\.com`),
+					},
+					FilterType: v1alpha1.FilterTypeExclude,
+				},
+			},
+			expectErr:     true,
+			expectedError: `exclude pattern (.*]\.abc\.com is invalid`,
+		},
+		{
+			name: "multiple regex exclude filter",
+			domainInput: []v1alpha1.ExternalDNSDomain{
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeRegex,
+						Pattern:   pointer.StringPtr(`(.*)\.abc\.com`),
+					},
+					FilterType: v1alpha1.FilterTypeExclude,
+				},
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeRegex,
+						Pattern:   pointer.StringPtr(`(.*)\.def\.com`),
+					},
+					FilterType: v1alpha1.FilterTypeExclude,
+				},
+			},
+			expectedArgs: []string{
+				`--regex-domain-exclusion=((.*)\.abc\.com)|((.*)\.def\.com)`,
+			},
+		},
+		{
+			name: "mixed domain filters",
+			domainInput: []v1alpha1.ExternalDNSDomain{
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeExact,
+						Name:      pointer.StringPtr("abc.com"),
+					},
+					FilterType: v1alpha1.FilterTypeInclude,
+				},
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeExact,
+						Name:      pointer.StringPtr("def.com"),
+					},
+					FilterType: v1alpha1.FilterTypeExclude,
+				},
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeRegex,
+						Pattern:   pointer.StringPtr(`(.*)\.ghi\.com`),
+					},
+					FilterType: v1alpha1.FilterTypeInclude,
+				},
+				{
+					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
+						MatchType: v1alpha1.DomainMatchTypeRegex,
+						Pattern:   pointer.StringPtr(`(.*)\.pqr\.com`),
+					},
+					FilterType: v1alpha1.FilterTypeExclude,
+				},
+			},
+			expectedArgs: []string{"--domain-filter=abc.com", "--exclude-domains=def.com", `--regex-domain-filter=(.*)\.ghi\.com`, `--regex-domain-exclusion=(.*)\.pqr\.com`},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			b := &externalDNSContainerBuilder{
+				externalDNS: &v1alpha1.ExternalDNS{
+					Spec: v1alpha1.ExternalDNSSpec{
+						Domains: tc.domainInput,
+					},
+				},
+			}
+			args, err := b.domainFilters()
+			if !tc.expectErr && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+			if tc.expectErr && err == nil {
+				t.Errorf("expected error, got nil")
+			}
+			if !tc.expectErr {
+				if !reflect.DeepEqual(args, tc.expectedArgs) {
+					t.Errorf("expected arguments %v, got %v", tc.expectedArgs, args)
+				}
+			} else {
+				if err == nil {
+					t.Errorf("expected error, got nil")
+					return
+				}
```

Comment:
This is redundant; line 246 already performs this check.

Reply:
Fixed